### PR TITLE
ROC-5433-Remove-unnecessary-fetch-call-before-save

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5433

### Change description

Remove fetch case by external id unnecessary before save.

### CODE REVIEW

File: src/main/features/claim/routes/pay.ts [hunk 1]
```diff
@@ -102,8 +74,30 @@ async function successHandler (res, next) {
...
     }
     await claimStoreClient.saveClaim(draft, user, features)
-  }
 
+  } catch (err) {
+    if (err.statusCode === HttpStatus.INTERNAL_SERVER_ERROR || err.statusCode === HttpStatus.SERVICE_UNAVAILABLE) {
+      logError(
+        user.id,
+        draft.document.claimant.payment,
+        `Payment processed successfully but there is problem saving claim from claim store externalId: ${externalId}.`
```
Comment (andrewwa-kainos) [inline src/main/features/claim/routes/pay.ts]: Can we fix the grammar please?
`... there was a problem saving claim '${externalId}' to the claim store`

File: src/main/features/claim/routes/pay.ts [hunk 2]
```diff
@@ -102,8 +74,30 @@ async function successHandler (res, next) {
...
+          payment: draft.document.claimant.payment,
+          error: err
+        })
+      next(err)
+      return
+    } else if (err.statusCode === HttpStatus.CONFLICT) {
+      logError(
+        user.id,
+        draft.document.claimant.payment,
+        `Payment processed successfully and claim already exist with same externalId: ${externalId}.`
```
Comment (andrewwa-kainos) [inline src/main/features/claim/routes/pay.ts]: Can we fix the grammar please?
`... and claim '${externalId}' already exists.`

File: src/main/features/claim/routes/pay.ts [hunk 3]
```diff
@@ -58,35 +58,7 @@ async function successHandler (res, next) {
   const user: User = res.locals.user
   const externalId: string = draft.document.externalId
 
-  let claimIsAlreadyFullyPersisted: boolean
-
   try {
```
Comment (KiranVarma02) [inline src/main/features/claim/routes/pay.ts]: Could i please ask, what is the reason earlier to check if claim is already fully persisted? and why we think its not necessary now? This is for my understanding...

File: src/main/features/claim/routes/pay.ts (hunk 2 above)
Comment (karthikkannankainos) [reply src/main/features/claim/routes/pay.ts]: done

File: src/main/features/claim/routes/pay.ts (hunk 1 above)
Comment (karthikkannankainos) [reply src/main/features/claim/routes/pay.ts]: Done @andrewwa-kainos .

File: src/main/features/claim/routes/pay.ts (hunk 3 above)
Comment (karthikkannankainos) [reply src/main/features/claim/routes/pay.ts]: @KiranVarma02 , we felt that making call is unnecessary as we are having transaction bound call.

Comment (KiranVarma02) [reply src/main/features/claim/routes/pay.ts]: ok, thanks for explaining

**Pull request merged**